Bottom: a600890d1ce33a8d974029b612b737256e44ccc7
Top:    97d24797024bf61c8da363fe3c81e98a532cb18d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:57:50 +0000

Make poly_Copy1 a static function internal to polygon1.c


---

diff --git a/src/polyarea.h b/src/polyarea.h
index be95bd2..fc6a63d 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -144,7 +144,6 @@ BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
 void poly_M_Incl(POLYAREA **list, POLYAREA *a);
 
 BOOLp poly_Copy0(POLYAREA **dst, const POLYAREA *src);
-BOOLp poly_Copy1(POLYAREA  *dst, const POLYAREA *src);
 
 BOOLp poly_InclContour(POLYAREA * p, PLINE * c);
 BOOLp poly_ExclContour(POLYAREA * p, PLINE * c);
diff --git a/src/polygon1.c b/src/polygon1.c
index 627447d..7847be6 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2930,20 +2930,7 @@ poly_CopyContour (PLINE ** dst, PLINE * src)
 /**********************************************************************/
 /* polygon routines */
 
-BOOLp
-poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
-{
-  *dst = NULL;
-  if (src != NULL)
-    *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
-    return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
-
-  return poly_Copy1 (*dst, src);
-}
-
-BOOLp
+static BOOLp
 poly_Copy1 (POLYAREA * dst, const POLYAREA * src)
 {
   PLINE *cur, **last = &dst->contours;
@@ -2954,13 +2941,27 @@ poly_Copy1 (POLYAREA * dst, const POLYAREA * src)
   for (cur = src->contours; cur != NULL; cur = cur->next)
     {
       if (!poly_CopyContour (last, cur))
-	return FALSE;
+        return FALSE;
       r_insert_entry (dst->contour_tree, (BoxType *) * last, 0);
       last = &(*last)->next;
     }
+
   return TRUE;
 }
 
+BOOLp
+poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
+{
+  *dst = NULL;
+  if (src != NULL)
+    *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
+  if (*dst == NULL)
+    return FALSE;
+  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
+
+  return poly_Copy1 (*dst, src);
+}
+
 void
 poly_M_Incl (POLYAREA ** list, POLYAREA * a)
 {
